python/dgl/contrib/cugraph/cugraph_storage.py

```diff
@@ -0,0 +1,220 @@
+# Copyright (c) 2019-2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: Requires cuGraph nightly cugraph-22.06.00a220417 or later
+
+import cugraph
+from cugraph.experimental import PropertyGraph
+from .cugraph_utils import cugraphToDGL
+import dgl
+import cupy
+import torch
+
+
+class CuGraphStorage(object):
+    """
+    Duck-typed version of the DGL GraphStorage class made for cuGraph
+    """
+    def __init__(self, g, ndata=None, edata=None):
+        # g must be a cugraph property graph
+        if not isinstance(g, PropertyGraph):
+            raise TypeError(
+                f"g must be a PropertyGraphh, got {type(g)}")
+
+        self.graphstore = cugraph.gnn.CuGraphStore(graph=g)
+        self._edata = self.graphstore.edata
+        self._ndata = self.graphstore.ndata
+
+    @property
+    def ndata(self):
+        return self._ndata
+
+    @property
+    def edata(self):
+        return self._edata
+
+    def get_node_storage(self, key, ntype=None):
+        node_col = self._ndata[self._ndata['_TYPE_'] == ntype][key]
```

Comment:
Rather than hardcode the internal column names, you should use `PropertyGraph.type_col_name` for safety.  This allows PropertyGraph to change the column names if it needs to without breaking clients:
```suggestion
        node_col = self._ndata[self._ndata[PropertyGraph.type_col_name] == ntype][key]
```
Also, instead of doing this every time `_TYPE_` is used, you could just define a constant and use that instead:
```
self.__pg_type = PropertyGraph.type_col_name
```